File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        minBuy = 10 * 10 * 10000

        for num in prices:
            minBuy = min(num, minBuy)
            maxProfit = max(maxProfit, num - minBuy)

        return maxProfit    

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = Integer.MIN_VALUE;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            }

            maxProfit = Math.max(maxProfit, price - minPrice);

        }

        return maxProfit;
        
        
    }
}